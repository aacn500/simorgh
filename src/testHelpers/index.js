import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ShallowRenderer from 'react-test-renderer/shallow';
import deepClone from 'ramda/src/clone';
import nodeLogger from '../app/lib/logger.node';

export const shouldMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const shallowRender = component => {
  const shallowRenderer = new ShallowRenderer();
  shallowRenderer.render(component);
  const result = shallowRenderer.getRenderOutput();

  return result;
};

export const shouldShallowMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = shallowRender(component);
    expect(tree).toMatchSnapshot();
  });
};

export const isNull = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toBeNull();
  });
};

export const setWindowValue = (key, value) => {
  const windowValue = window[key];
  delete window[key];

  let newValue = value;

  if (value && typeof value === 'object') {
    newValue = {
      ...deepClone(windowValue),
      ...value,
    };
  }

  Object.defineProperty(window, key, {
    value: newValue,
    writable: true,
  });
};

export const resetWindowValue = (key, value) => {
  Object.defineProperty(window, key, {
    value,
    writable: true,
  });
};

const mocks = {
  error: jest.fn(),
  warn: jest.fn(),
  info: jest.fn(),
  verbose: jest.fn(),
  debug: jest.fn(),
  silly: jest.fn(),
};

jest.mock('../app/lib/logger.node', () => jest.fn());
nodeLogger.mockImplementation(() => mocks);

export const loggerMock = mocks;
