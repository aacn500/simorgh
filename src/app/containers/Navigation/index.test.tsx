import React from 'react';
import { render } from '@testing-library/react';
import { shouldShallowMatchSnapshot } from '../../../testHelpers';
import NavigationContainer from './index';
import igboConfig from '../../lib/config/services/igbo';

jest.mock('react', () => {
  const original = jest.requireActual('react');
  return {
    ...original,
    useContext: jest.fn(),
  };
});

const { useContext } = jest.requireMock('react');

describe('Navigation Container', () => {
  beforeEach(() => {
    useContext.mockReturnValue(igboConfig);
  });

  afterEach(() => {
    useContext.mockReset();
  });

  describe('snapshots', () => {
    shouldShallowMatchSnapshot(
      'should render a Navigation with igbo links correctly',
      <NavigationContainer />,
    );
  });

  describe('assertions', () => {
    it('should render a Navigation with a Skip to content link, linking to #content', () => {
      const { container } = render(<NavigationContainer />);

      const skipLink = container.querySelector('a') as HTMLAnchorElement;
      const skipLinkHref = skipLink.getAttribute('href');

      expect(skipLinkHref).toBe('#content');
    });
  });
});
