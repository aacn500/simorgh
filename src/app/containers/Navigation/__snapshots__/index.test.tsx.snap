// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Container snapshots should render a Navigation with igbo links correctly 1`] = `
<Navigation
  script={
    Object {
      "bodyCopy": Object {
        "groupA": Object {
          "fontSize": "15",
          "lineHeight": "20",
        },
        "groupB": Object {
          "fontSize": "16",
          "lineHeight": "22",
        },
        "groupD": Object {
          "fontSize": "16",
          "lineHeight": "22",
        },
      },
      "brevier": Object {
        "groupA": Object {
          "fontSize": "14",
          "lineHeight": "18",
        },
        "groupB": Object {
          "fontSize": "14",
          "lineHeight": "18",
        },
        "groupD": Object {
          "fontSize": "13",
          "lineHeight": "16",
        },
      },
      "canon": Object {
        "groupA": Object {
          "fontSize": "28",
          "lineHeight": "32",
        },
        "groupB": Object {
          "fontSize": "32",
          "lineHeight": "36",
        },
        "groupD": Object {
          "fontSize": "44",
          "lineHeight": "48",
        },
      },
      "doublePica": Object {
        "groupA": Object {
          "fontSize": "18",
          "lineHeight": "22",
        },
        "groupB": Object {
          "fontSize": "20",
          "lineHeight": "24",
        },
        "groupD": Object {
          "fontSize": "24",
          "lineHeight": "28",
        },
      },
      "greatPrimer": Object {
        "groupA": Object {
          "fontSize": "18",
          "lineHeight": "22",
        },
        "groupB": Object {
          "fontSize": "18",
          "lineHeight": "22",
        },
        "groupD": Object {
          "fontSize": "20",
          "lineHeight": "24",
        },
      },
      "longPrimer": Object {
        "groupA": Object {
          "fontSize": "15",
          "lineHeight": "18",
        },
        "groupB": Object {
          "fontSize": "15",
          "lineHeight": "18",
        },
        "groupD": Object {
          "fontSize": "14",
          "lineHeight": "18",
        },
      },
      "minion": Object {
        "groupA": Object {
          "fontSize": "12",
          "lineHeight": "16",
        },
        "groupB": Object {
          "fontSize": "12",
          "lineHeight": "16",
        },
        "groupD": Object {
          "fontSize": "12",
          "lineHeight": "16",
        },
      },
      "paragon": Object {
        "groupA": Object {
          "fontSize": "20",
          "lineHeight": "24",
        },
        "groupB": Object {
          "fontSize": "22",
          "lineHeight": "26",
        },
        "groupD": Object {
          "fontSize": "28",
          "lineHeight": "32",
        },
      },
      "pica": Object {
        "groupA": Object {
          "fontSize": "15",
          "lineHeight": "20",
        },
        "groupB": Object {
          "fontSize": "16",
          "lineHeight": "20",
        },
        "groupD": Object {
          "fontSize": "16",
          "lineHeight": "20",
        },
      },
      "trafalgar": Object {
        "groupA": Object {
          "fontSize": "20",
          "lineHeight": "24",
        },
        "groupB": Object {
          "fontSize": "24",
          "lineHeight": "28",
        },
        "groupD": Object {
          "fontSize": "32",
          "lineHeight": "36",
        },
      },
    }
  }
  service="igbo"
  skipLinkText="Wụga n’ọdịnaya"
>
  <NavigationUl>
    <NavigationLi
      active={true}
      currentPageText="Peegi ị nọ ugbua"
      dir="ltr"
      script={
        Object {
          "bodyCopy": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "20",
            },
            "groupB": Object {
              "fontSize": "16",
              "lineHeight": "22",
            },
            "groupD": Object {
              "fontSize": "16",
              "lineHeight": "22",
            },
          },
          "brevier": Object {
            "groupA": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
            "groupB": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
            "groupD": Object {
              "fontSize": "13",
              "lineHeight": "16",
            },
          },
          "canon": Object {
            "groupA": Object {
              "fontSize": "28",
              "lineHeight": "32",
            },
            "groupB": Object {
              "fontSize": "32",
              "lineHeight": "36",
            },
            "groupD": Object {
              "fontSize": "44",
              "lineHeight": "48",
            },
          },
          "doublePica": Object {
            "groupA": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupB": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupD": Object {
              "fontSize": "24",
              "lineHeight": "28",
            },
          },
          "greatPrimer": Object {
            "groupA": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupB": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupD": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
          },
          "longPrimer": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "18",
            },
            "groupB": Object {
              "fontSize": "15",
              "lineHeight": "18",
            },
            "groupD": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
          },
          "minion": Object {
            "groupA": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
            "groupB": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
            "groupD": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
          },
          "paragon": Object {
            "groupA": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupB": Object {
              "fontSize": "22",
              "lineHeight": "26",
            },
            "groupD": Object {
              "fontSize": "28",
              "lineHeight": "32",
            },
          },
          "pica": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "20",
            },
            "groupB": Object {
              "fontSize": "16",
              "lineHeight": "20",
            },
            "groupD": Object {
              "fontSize": "16",
              "lineHeight": "20",
            },
          },
          "trafalgar": Object {
            "groupA": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupB": Object {
              "fontSize": "24",
              "lineHeight": "28",
            },
            "groupD": Object {
              "fontSize": "32",
              "lineHeight": "36",
            },
          },
        }
      }
      service="igbo"
      url="/igbo"
    >
      Akụkọ
    </NavigationLi>
    <NavigationLi
      active={false}
      currentPageText="Peegi ị nọ ugbua"
      dir="ltr"
      script={
        Object {
          "bodyCopy": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "20",
            },
            "groupB": Object {
              "fontSize": "16",
              "lineHeight": "22",
            },
            "groupD": Object {
              "fontSize": "16",
              "lineHeight": "22",
            },
          },
          "brevier": Object {
            "groupA": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
            "groupB": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
            "groupD": Object {
              "fontSize": "13",
              "lineHeight": "16",
            },
          },
          "canon": Object {
            "groupA": Object {
              "fontSize": "28",
              "lineHeight": "32",
            },
            "groupB": Object {
              "fontSize": "32",
              "lineHeight": "36",
            },
            "groupD": Object {
              "fontSize": "44",
              "lineHeight": "48",
            },
          },
          "doublePica": Object {
            "groupA": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupB": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupD": Object {
              "fontSize": "24",
              "lineHeight": "28",
            },
          },
          "greatPrimer": Object {
            "groupA": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupB": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupD": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
          },
          "longPrimer": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "18",
            },
            "groupB": Object {
              "fontSize": "15",
              "lineHeight": "18",
            },
            "groupD": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
          },
          "minion": Object {
            "groupA": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
            "groupB": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
            "groupD": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
          },
          "paragon": Object {
            "groupA": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupB": Object {
              "fontSize": "22",
              "lineHeight": "26",
            },
            "groupD": Object {
              "fontSize": "28",
              "lineHeight": "32",
            },
          },
          "pica": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "20",
            },
            "groupB": Object {
              "fontSize": "16",
              "lineHeight": "20",
            },
            "groupD": Object {
              "fontSize": "16",
              "lineHeight": "20",
            },
          },
          "trafalgar": Object {
            "groupA": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupB": Object {
              "fontSize": "24",
              "lineHeight": "28",
            },
            "groupD": Object {
              "fontSize": "32",
              "lineHeight": "36",
            },
          },
        }
      }
      service="igbo"
      url="/igbo/egwuregwu"
    >
      Egwuregwu
    </NavigationLi>
    <NavigationLi
      active={false}
      currentPageText="Peegi ị nọ ugbua"
      dir="ltr"
      script={
        Object {
          "bodyCopy": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "20",
            },
            "groupB": Object {
              "fontSize": "16",
              "lineHeight": "22",
            },
            "groupD": Object {
              "fontSize": "16",
              "lineHeight": "22",
            },
          },
          "brevier": Object {
            "groupA": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
            "groupB": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
            "groupD": Object {
              "fontSize": "13",
              "lineHeight": "16",
            },
          },
          "canon": Object {
            "groupA": Object {
              "fontSize": "28",
              "lineHeight": "32",
            },
            "groupB": Object {
              "fontSize": "32",
              "lineHeight": "36",
            },
            "groupD": Object {
              "fontSize": "44",
              "lineHeight": "48",
            },
          },
          "doublePica": Object {
            "groupA": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupB": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupD": Object {
              "fontSize": "24",
              "lineHeight": "28",
            },
          },
          "greatPrimer": Object {
            "groupA": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupB": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupD": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
          },
          "longPrimer": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "18",
            },
            "groupB": Object {
              "fontSize": "15",
              "lineHeight": "18",
            },
            "groupD": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
          },
          "minion": Object {
            "groupA": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
            "groupB": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
            "groupD": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
          },
          "paragon": Object {
            "groupA": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupB": Object {
              "fontSize": "22",
              "lineHeight": "26",
            },
            "groupD": Object {
              "fontSize": "28",
              "lineHeight": "32",
            },
          },
          "pica": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "20",
            },
            "groupB": Object {
              "fontSize": "16",
              "lineHeight": "20",
            },
            "groupD": Object {
              "fontSize": "16",
              "lineHeight": "20",
            },
          },
          "trafalgar": Object {
            "groupA": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupB": Object {
              "fontSize": "24",
              "lineHeight": "28",
            },
            "groupD": Object {
              "fontSize": "32",
              "lineHeight": "36",
            },
          },
        }
      }
      service="igbo"
      url="/igbo/media/video"
    >
      Ihe nkiri
    </NavigationLi>
    <NavigationLi
      active={false}
      currentPageText="Peegi ị nọ ugbua"
      dir="ltr"
      script={
        Object {
          "bodyCopy": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "20",
            },
            "groupB": Object {
              "fontSize": "16",
              "lineHeight": "22",
            },
            "groupD": Object {
              "fontSize": "16",
              "lineHeight": "22",
            },
          },
          "brevier": Object {
            "groupA": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
            "groupB": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
            "groupD": Object {
              "fontSize": "13",
              "lineHeight": "16",
            },
          },
          "canon": Object {
            "groupA": Object {
              "fontSize": "28",
              "lineHeight": "32",
            },
            "groupB": Object {
              "fontSize": "32",
              "lineHeight": "36",
            },
            "groupD": Object {
              "fontSize": "44",
              "lineHeight": "48",
            },
          },
          "doublePica": Object {
            "groupA": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupB": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupD": Object {
              "fontSize": "24",
              "lineHeight": "28",
            },
          },
          "greatPrimer": Object {
            "groupA": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupB": Object {
              "fontSize": "18",
              "lineHeight": "22",
            },
            "groupD": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
          },
          "longPrimer": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "18",
            },
            "groupB": Object {
              "fontSize": "15",
              "lineHeight": "18",
            },
            "groupD": Object {
              "fontSize": "14",
              "lineHeight": "18",
            },
          },
          "minion": Object {
            "groupA": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
            "groupB": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
            "groupD": Object {
              "fontSize": "12",
              "lineHeight": "16",
            },
          },
          "paragon": Object {
            "groupA": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupB": Object {
              "fontSize": "22",
              "lineHeight": "26",
            },
            "groupD": Object {
              "fontSize": "28",
              "lineHeight": "32",
            },
          },
          "pica": Object {
            "groupA": Object {
              "fontSize": "15",
              "lineHeight": "20",
            },
            "groupB": Object {
              "fontSize": "16",
              "lineHeight": "20",
            },
            "groupD": Object {
              "fontSize": "16",
              "lineHeight": "20",
            },
          },
          "trafalgar": Object {
            "groupA": Object {
              "fontSize": "20",
              "lineHeight": "24",
            },
            "groupB": Object {
              "fontSize": "24",
              "lineHeight": "28",
            },
            "groupD": Object {
              "fontSize": "32",
              "lineHeight": "36",
            },
          },
        }
      }
      service="igbo"
      url="/igbo/popular/read"
    >
      Nke ka ewuewu
    </NavigationLi>
  </NavigationUl>
</Navigation>
`;
